compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -3625,8 +3625,9 @@ class Typer(@constructorOnly nestingLevel: Int = 0) extends Namer
       val remembered = // report AmbiguousReferences as priority, otherwise last error
         (errs.filter(_.msg.isInstanceOf[AmbiguousReference]) ++ errs).take(1)
       for err <- remembered do
+        val tree = if app.isEmpty then qual else app
         rememberSearchFailure(qual,
-          SearchFailure(app.withType(FailedExtension(app, selectionProto, err.msg))))
+          SearchFailure(tree.withType(FailedExtension(tree, selectionProto, err.msg))))
```

Comment:
should I move it to a separate PR then - I forced unpickling to always throw an error and then this crashed because sometimes `app` could be the EmptyTree, so assigning type does nothing

Reply:
If it's necessary for this PR to pass the tests, then keep it here. Otherwise move it in a separate PR, yes.

Reply:
dropped the change, now its in https://github.com/lampepfl/dotty/pull/18830